.github/workflows/codec.yml

```diff
@@ -0,0 +1,24 @@
+name: cppcheck
+
+on:
+  push
+jobs:
+  codec:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Installing cppcheck
+        run: sudo apt-get install cppcheck 
+      - name: Permission
+        run: chmod a+x ./cppcheck.sh
+      - name: Code analysis
+        run: ./cppcheck.sh
+      - name: Artifact performance
+        uses: actions/upload-artifact@v2
+        with:
+          path: ./performance_warnings.txt
```

Comment:
a scriptben `cppcheck_performance.txt` a fájlnév, és nincs külön performance meg style kimenet (nem is kell különszedni)
amúgy az ilyenek miatt érdemes kerülni a nevek duplikálását, pl. megkaphatná a bash script argumentumként, és akkor nem kellene beleírni literálként